src/Tests/CustomCacheTagsTest.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\views_custom_cache_tag\CustomCacheTagsTest.
+ */
+
+namespace Drupal\views_custom_cache_tag\Tests;
+
+use Drupal\Component\Utility\String;
+use Drupal\Core\Entity\Entity;
+use Drupal\Core\Url;
+use Drupal\node\Entity\Node;
+use Drupal\simpletest\WebTestBase;
+use Drupal\system\Tests\Cache\AssertPageCacheTagsTrait;
+use Drupal\views\Entity\View;
+
+/**
+ * Tests the custom cache tags in views.
+ *
+ * @group views_custom_cache_tag
+ */
+class CustomCacheTagsTest extends WebTestBase {
+
+  use AssertPageCacheTagsTrait;
+
+  /**
+   * Disabled config schema checking temporarily until all errors are resolved.
+   */
+  protected $strictConfigSchema = FALSE;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = array(
+    'node',
+    'views',
+    'views_custom_cache_tag_demo'
+  );
+
+  /**
+   * Tests the cache tag invalidation.
+   */
+  public function testCustomCacheTags() {
+
+    $this->enablePageCaching();
+    // Create a new node of type A.
+    $node_a = Node::create([
+      'body' => [
+        [
+          'value' => $this->randomMachineName(32),
+          'format' => filter_default_format(),
+        ]
+      ],
+      'type' => 'node_type_a',
+      'created' => 1,
+      'title' => $this->randomMachineName(8),
+      'nid' => 2,
+            ]);
+    $node_a->enforceIsNew(TRUE);
+    $node_a->save();
+
+    // Create a new node of type B.
+    $node_b = Node::create([
+      'body' => [
+        [
+          'value' => $this->randomMachineName(32),
+          'format' => filter_default_format(),
+        ]
+      ],
+      'type' => 'node_type_b',
+      'created' => 1,
+      'title' => $this->randomMachineName(8),
+      'nid' => 3,
+    ]);
+    $node_b->enforceIsNew(TRUE);
+    $node_b->save();
+
+    // Check the cache tags in the views.
+    $this->assertPageCacheTags(Url::fromRoute('view.view_node_type_a.page_1'), array(
```

Comment:
surprised that we don't also have the cache tags of the node we're viewing here. what kind of view is this?